### ISSUE TITLE
False-Positive for "is_numeric"

### ISSUE THREAD

**Issue by voku** [kalessil/phpinspectionsea#1009]
| Subject    | Details                                                  |
| :--------- | :------------------------------------------------------- |
| Issue type | False-Positive     |
| IDE        | PhpStorm: PhpStorm 2018.2.2 (Build #PS-182.4129.45)      |
| Plugin     | Php Inspections (EA Extended Ultimate) 2.0.7 |

Description:
--
```is_numeric($string)```: is reported as "Makes no sense, because of parameter type declaration.", but this function accept string or int, so this should be ok?

```php
  /**
   * @param string $str
   *
   * @return bool
   */
  public static function to_boolean(string $str): bool
  {
    $key = \strtolower($str);

    // Info: http://php.net/manual/en/filter.filters.validate.php
    $map = [
        'true'  => true,
        '1'     => true,
        'on'    => true,
        'yes'   => true,
        'false' => false,
        '0'     => false,
        'off'   => false,
        'no'    => false,
    ];

    if (isset($map[$key])) {
      return $map[$key];
    }

    if (\is_numeric($str)) {
      return $str + 0 > 0;
    }

    return (bool)self::trim($str);
  }
```

Environment details
---
PhpStorm 2018.2.2
Build #PS-182.4129.45, built on August 22, 2018
Licensed to Portable UTF-8 /  
Subscription is active until April 5, 2019
For non-commercial open source development only.
JRE: 1.8.0_152-release-1248-b8 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.4.0-133-generic

**Comment by kalessil** [comment]
Will check, looks like a type identification bug.

**Comment by kalessil** [comment]
Fixed!
